crates/core/tests/dom.rs

```diff
@@ -61,6 +61,21 @@ fn dom_builder_example() {
     assert_eq!(buffer.as_str(), expected);
 }
 
+#[test]
+fn iframe_closing_tag_roundtrip() {
```

Comment:
Does this test fail with the previous parser logic?

Reply:
indeed it does. the two closing tags are treated as strings and then the parser tries to correct by closing off the parent tags.